### PR TITLE
Use solidity custom errors

### LINKED ISSUE #132: Remove `require` statements in favor of Solidity Errors
All the instances that now the `Ethernauts` contract use the `require()` statement should use custom Errors with `reverts`.

More info: https://docs.soliditylang.org/en/v0.8.10/contracts.html?highlight=errors#errors-and-the-revert-statement

This issue will also require to update all the necessary tests.

### PR DESCRIPTION
**Pull request by mariobyn**
Closes #132

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,5 @@
 node_modules
 coverage*
 
-docker/data
\ No newline at end of file
+docker/data
+package-lock.json
```
Comment (slorenzo) [inline .gitignore:7]: It shouldn’t be here, wee need package-lock.json

File: packages/hardhat/package.json [hunk 2]
```diff
@@ -29,8 +29,10 @@
     "cids": "1.1.9",
     "dotenv": "10.0.0",
     "envalid": "7.2.1",
+    "hardhat": "2.8.0",
     "ipfs-http-client": "52.0.3",
-    "ipfs-only-hash": "4.0.0"
+    "ipfs-only-hash": "4.0.0",
+    "prettier": "2.5.1"
```
Comment (slorenzo) [inline packages/hardhat/package.json:35]: Prettier is already installed for the whole project. Look at the main package.json

**Review by slorenzo — CHANGES_REQUESTED** [review]

File: packages/hardhat/test/Recovery.test.js [hunk 3]
```diff
@@ -51,7 +51,7 @@ describe('Recovery', () => {
       assert.notEqual(balance, 0);
       await assertRevert(
         Ethernauts.connect(owner).recoverTokens(Token.address, Token.address, balance),
-        'Invalid Address'
+        'RecoverTokenError("' + Token.address + '", "' + Token.address + '")'
```
Comment (karmacoma-eth) [inline packages/hardhat/test/Recovery.test.js:54]: 🥳

**Review by karmacoma-eth — APPROVED** [review]

File: packages/hardhat/test/Recovery.test.js (hunk 3 above)
Comment (mariobyn) [reply packages/hardhat/test/Recovery.test.js:54]: but for some reason the CI fails and I am not sure why

**Review by slorenzo — APPROVED** [review]

**Pull request merged**